server/src/main/java/org/apache/druid/server/coordinator/SegmentCountsPerInterval.java

```diff
@@ -61,6 +66,11 @@ public long getTotalSegmentBytes()
     return totalSegmentBytes;
   }
 
+  public Set<DataSegment> getSegments()
```

Comment:
@adarshsanjeev , I took a look at the code and I feel there is another way to do this instead of maintaining a new set here.

- Rename the `ServerHolder.getProjectedSegments()` method to `ServerHolder.getProjectedSegmentCounts()`
- Add a new method `ServerHolder.getProjectedSegments()`
- This way the methods `isProjectedSegment()` and `getProjectedSegments()` will be aligned with each other better.

```java
public Set<DataSegment> getProjectedSegments()
{
    final Set<DataSegment> projectedSegments = new HashSet<>(getServedSegments());
    queuedSegments.forEach((segment, action) -> {
        if (action.isLoad()) {
            projectedSegments.add(segment);
        }
    });
    
    return projectedSegments;
}
```

Let me know if this makes sense.

Reply:
I think this makes sense. We would also need to account for dropped segments, so it would need something like
```java
if (action.isLoad()) {
  projectedSegments.add(segment);
} else {
  projectedSegments.remove(segment);
}
```
as well.

Reply:
you might need to do 

```
else if (action == DROP)
```

since the action can be MOVE_FROM too.